### PR TITLE
Add 0.30.1 kapp-controller package

### LINKED ISSUE #3296: Bump Kapp-Controller package to 0.30.1
## Feature Request
Bump Kapp-Controller package to 0.30.1

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

## What this PR does / why we need it

* Add kapp-controller 0.30.1
* Add write access to /etc/pki/tls/certs/ca-bundle.crt so kapp-controller will have no issue in writing custom certificates into it.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Release Kapp-Controller package 0.30.1
```

## Which issue(s) this PR fixes

Fixes: #3296

### CODE REVIEW

File: addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/images.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+apiVersion: imgpkg.carvel.dev/v1alpha1
+images:
+- annotations:
+    kbld.carvel.dev/id: ghcr.io/vmware-tanzu/carvel-kapp-controller@sha256:4141b93108b1f3431e786638a115a7e0876d756e7a13acfddd7aaa3bae41e202
```
Comment (blc1996) [inline addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/images.yml:5]: Note for reviewer: Image update for 0.30.1,
was `ghcr.io/vmware-tanzu/carvel-kapp-controller@sha256:46f9c4e78d80a322ae3159cb97069350b445b974664f4aead0ab4ad593d79f45` for 0.30.0

File: addons/packages/kapp-controller/0.30.1/bundle/config/values.star [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+else:
+    kappNamespace = values.namespace
+end
+
+def generateBashCmdForDNS(coreDNSIP):
+    # This command added the coreDNS IP as the first entry of resolv.conf
+    # In this way, Kapp Controller will have cluster IP access,
+    # and still able to resolve enternal urls when core DNS is unavailable
+
+    return "cp /etc/resolv.conf /etc/resolv.conf.bak; sed '1 i nameserver " + coreDNSIP + "' /etc/resolv.conf.bak > /etc/resolv.conf; rm /etc/resolv.conf.bak; cp -R /etc/* /kapp-etc; chmod g+w /kapp-etc/pki/tls/certs/ca-bundle.crt && chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt"
```
Comment (blc1996) [inline addons/packages/kapp-controller/0.30.1/bundle/config/values.star:17]: Note for reviewer: `chmod g+w /kapp-etc/pki/tls/certs/ca-bundle.crt && chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt` is added for cert file write access.

File: addons/packages/kapp-controller/0.30.1/package.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      template:
+        - ytt:
+            paths:
+              - config/
+        - kbld:
+            paths:
+              - "-"
+              - .imgpkg/images.yml
+      deploy:
+        - kapp: {}
```
Comment (blc1996) [inline addons/packages/kapp-controller/0.30.1/package.yaml:25]: Note for reviewer: `schema.yaml` file is removed to minimize risk

**Review by blc1996 — COMMENTED** [review]
Some notes for reviewers:
The only change from 0.30.0 to 0.30.1 in upstream template is the image url and some CRD field descriptions.
Therefore, we nothing in 0.30.0 overlays need to be changed, except for the image url and bug fix.

File: addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/bundle.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: Bundle
+metadata:
+  name: kapp-controller
+authors:
+- name: Daniel Guo
```
Comment (shivaani0505) [inline addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/bundle.yml:6]: Here the name and email should be yours as you are making the change

**Comment by shivaani0505** [comment]
Except for the minor comment, changes look good to me @blc1996

**Comment by shivaani0505** [comment]
Could you create an issue for this change in community edition and update that in Fixes: section @blc1996

File: addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/bundle.yml (hunk 4 above)
Comment (blc1996) [reply addons/packages/kapp-controller/0.30.1/bundle/.imgpkg/bundle.yml:6]: Added my name

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Pull request merged**